pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,207 @@
+	eventRecorder events.Recorder,
+) factory.Controller {
+
+	ret := &LatencyProfileController{
+		operatorClient:  operatorClient,
+		configClient:    configClient,
+		configMapClient: v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		nodeLister:      nodeInformer.Lister(),
+	}
+
+	return factory.New().WithInformers(
+		// this is for our general configuration input and our status output in case another actor changes it
+		operatorClient.Informer(),
+
+		// We use nodeInformer for observing current worker latency profile
+		nodeInformer.Informer(),
+
+		// for configmaps of operator client target namespace
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Informer(),
+	).ResyncEvery(time.Minute).WithSync(ret.sync).ToController(
+		"LatencyProfileController",
+		eventRecorder.WithComponentSuffix("latency-profile-controller"),
+	)
+}
+
+func (c *LatencyProfileController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	// ensures that the given worker latency profile have synced their respective kcm arguments
+	// to all control plane nodes in the cluster
+	syncErr := func() error {
+		_, operatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+		if err != nil {
+			return err
+		}
+
+		// Collect the unique set of revisions of the node static pods
+		revisionMap := map[int32]struct{}{}
+		uniqueRevisions := []int32{}
+		for _, nodeStatus := range operatorStatus.NodeStatuses {
+			revision := nodeStatus.CurrentRevision
+			if _, ok := revisionMap[revision]; !ok {
+				revisionMap[revision] = struct{}{}
+				uniqueRevisions = append(uniqueRevisions, revision)
+			}
+		}
+
+		// Collect the current latency profile
+		configNodeObj, err := c.nodeLister.Get("cluster")
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		} else if errors.IsNotFound(err) {
+			// if config/v1/node/cluster object is not found this controller should do nothing
+			return nil
+		}
+
+		desiredControllerManagerArgumentVals := map[string]string{}
+		switch configNodeObj.Spec.WorkerLatencyProfile {
+		case apiconfigv1.DefaultUpdateDefaultReaction:
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.DefaultNodeMonitorGracePeriod))
+		case apiconfigv1.MediumUpdateAverageReaction:
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.MediumNodeMonitorGracePeriod))
+		case apiconfigv1.LowUpdateSlowReaction:
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.LowNodeMonitorGracePeriod))
+		}
+
+		condition := metav1.Condition{
+			Type:   apiconfigv1.KubeControllerManagerDegraded,
+			Status: metav1.ConditionUnknown,
+		}
+		// do nothing if workerLatencyProfile is not set
+		if configNodeObj.Spec.WorkerLatencyProfile == "" {
+			// TODO: in case worker latency profile is transitioned
```

Comment:
I would say the KCM is not Degraded here and the status should be False

Reply:
Addressing, it similar to: https://github.com/openshift/cluster-kube-controller-manager-operator/pull/611#discussion_r853274606